sstables/mx/writer.cc

```diff
@@ -1470,7 +1467,13 @@ void writer::consume_end_of_stream() {
     _sst.write_compression(_pc);
     auto features = sstable_enabled_features::all();
     run_identifier identifier{_run_identifier};
-    std::optional<scylla_metadata::large_data_stats> ld_stats(std::move(_large_data_stats));
+    scylla_metadata::large_data_stats lds;
+    lds.map.emplace(large_data_type::partition_size, std::move(_partition_size_entry));
+    lds.map.emplace(large_data_type::row_size, std::move(_row_size_entry));
+    lds.map.emplace(large_data_type::cell_size, std::move(_cell_size_entry));
+    lds.map.emplace(large_data_type::rows_in_partition, std::move(_rows_in_partition_entry));
+    lds.map.emplace(large_data_type::elements_in_collection, std::move(_elements_in_collection_entry));
+    std::optional<scylla_metadata::large_data_stats> ld_stats(std::move(lds));
```

Comment:
nit: why not make it an optional in the first place?

Reply:
hmm, yes, this can be done.
my first version called the automatic variable `_large_data_stats` to mimic the previous member so to reduce the code churn, but then I decided to rename it, voiding the churn benefits  so you're right, we can just populate ld_stats after it's created empty.